diff -ur support/shobj-conf support/shobj-conf
--- support/shobj-conf	2009-10-03 01:00:36.000000000 +0200
+++ support/shobj-conf	2009-10-03 01:12:01.000000000 +0200
@@ -145,18 +145,18 @@
 darwin[89]*)
 	SHOBJ_STATUS=supported
 	SHLIB_STATUS=supported
-	
+
 	SHOBJ_CFLAGS='-fno-common'
 
-	SHOBJ_LD='MACOSX_DEPLOYMENT_TARGET=10.3 ${CC}'
+	SHOBJ_LD='${CC}'
 
 	SHLIB_LIBVERSION='$(SHLIB_MAJOR)$(SHLIB_MINOR).$(SHLIB_LIBSUFF)'
 	SHLIB_LIBSUFF='dylib'
 
-	SHOBJ_LDFLAGS='-dynamiclib -dynamic -undefined dynamic_lookup -arch_only `/usr/bin/arch`'
-	SHLIB_XLDFLAGS='-dynamiclib -arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
+	SHOBJ_LDFLAGS='-dynamiclib -dynamic -undefined dynamic_lookup'
+	SHLIB_XLDFLAGS='-dynamiclib -install_name $(libdir)/${@:$(SHLIB_MAJOR)$(SHLIB_MINOR).$(SHLIB_LIBSUFF)=$(SHLIB_MAJOR).$(SHLIB_LIBSUFF)} -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR)  -compatibility_version $(SHLIB_MAJOR) -v -dynamiclib'
 
-	SHLIB_LIBS='-lncurses'	# see if -lcurses works on MacOS X 10.1 
+	SHLIB_LIBS='-lncurses'	# see if -lcurses works on MacOS X 10.1
 	;;
 
 darwin*|macosx*)
@@ -172,14 +172,14 @@
 
 	case "${host_os}" in
 	darwin[789]*)	SHOBJ_LDFLAGS=''
-			SHLIB_XLDFLAGS='-dynamiclib -arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
+			SHLIB_XLDFLAGS='-dynamiclib -install_name $(libdir)/${@:$(SHLIB_MAJOR)$(SHLIB_MINOR).$(SHLIB_LIBSUFF)=$(SHLIB_MAJOR).$(SHLIB_LIBSUFF)} -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR)  -compatibility_version $(SHLIB_MAJOR) -v -dynamiclib'
 			;;
 	*)		SHOBJ_LDFLAGS='-dynamic'
-			SHLIB_XLDFLAGS='-arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
+			SHLIB_XLDFLAGS='-install_name $(libdir)/${@:$(SHLIB_MAJOR)$(SHLIB_MINOR).$(SHLIB_LIBSUFF)=$(SHLIB_MAJOR).$(SHLIB_LIBSUFF)} -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v -dynamiclib'
 			;;
 	esac
 
-	SHLIB_LIBS='-lncurses'	# see if -lcurses works on MacOS X 10.1 
+	SHLIB_LIBS='-lncurses'	# see if -lcurses works on MacOS X 10.1
 	;;
 
 openbsd*|netbsd*)
